### ISSUE TITLE
Parser parses random strings to a date

### ISSUE THREAD

**Issue by asishm** [dateutil/dateutil#440]
```py
>>> parser.parse('2.3, 4.1, 4.3')
datetime.datetime(2004, 2, 4, 0, 0)

>>> parser.parse('6.2 . . . 6.5 6.6')
datetime.datetime(2006, 6, 6, 0, 0)
```

I'm not sure why the parser is parsing these strings to datetimes. 

The way I get these strings is I have a bunch of text to which I throw in regular expressions (similar to the `datefinder` library). And use these extracted strings and pass it in to the parser. If it errors out, I claim there isn't a date, but I've been finding strings like these where it parses to a date that it shouldn't.

**Comment by jbrockmendel** [comment]
You're right that the parser is too loose for a lot of these random-text cases.  These two cases are a good start.  Do you happen to have a big pile of examples that we could turn into test cases?

**Comment by pganssle** [comment]
I'm not sure that "throw something at the parser and see if it's a date" is really going to be *terribly* fruitful, because the parser tries to err *way* on the side of false positives - if it sees anything that could reasonably be a date, it will give you a datetime.

That said, I think the problem here is that these strings are made entirely of valid separators and valid values for date *or* time, but we're not keeping track of which ones get used (or possibly as it walks through the string it finds new values and silently overwrites the original ones it picked up), and the fact that the separators don't actually *separate* anything is not being picked up. I imagine this will partly get solved when we start enforcing that we shouldn't find two answers for the same value, but if that's the problem, why is the "3" in `4.3` getting ignored? It's certainly not getting overwritten by anything later..

**Comment by asishm** [comment]
I don't have something with me right now. 

I may be able to come up with some examples later. Most of my examples will be from pdfs that have been converted to text and are biased towards regulatory documents available in the public domain.

@pganssle I agree (about throwing it into the parser, but these are not really random strings. Most patterns are similar to the ones found https://github.com/akoumjian/datefinder/blob/master/datefinder.py customizing to the documents that I have), which is why I'm going to try doing some further cleaning on my end before throwing it into the parser. I think these ones are fairly easy to solve on my end, but I'm going to have to spend a bit more time looking at my entire dataset to see if there's any other examples that stand out.

**Comment by jbrockmendel** [comment]
> I may be able to come up with some examples later.

@asishm No hurry.  The best-case here would be if you have a pile of examples with a mix between is-a-date and not-a-date.

> the parser tries to err way on the side of false positives - if it sees anything that could reasonably be a date, it will give you a datetime.

@pganssle That's definitely the case now, but I think `'6.2 . . . 6.5 6.6'` is on the far side of "could reasonably be a date".  Regardless of where the line for "reasonable" is eventually drawn, we'll want/need a thorough set of test cases on both sides of it.
